client/my-sites/site-settings/media-settings.jsx

```diff
@@ -260,7 +249,11 @@ export default connect(
 		const siteInDevMode = isJetpackSiteInDevelopmentMode( state, selectedSiteId );
 		const sitePlanSlug = getSitePlanSlug( state, selectedSiteId );
 		const moduleUnavailableInDevMode = isJetpackModuleUnavailableInDevelopmentMode( state, selectedSiteId, 'photon' );
-		const isVideoPressAvailable = includes( plansIncludingVideoPress, sitePlanSlug );
+		const isVideoPressAvailable = (
+			hasFeature( state, selectedSiteId, FEATURE_VIDEO_UPLOADS ) ||
+			hasFeature( state, selectedSiteId, FEATURE_VIDEO_UPLOADS_JETPACK_PREMIUM ) ||
+			hasFeature( state, selectedSiteId, FEATURE_VIDEO_UPLOADS_JETPACK_PRO )
+		);
```

Comment:
I considered this approach and explored initial implementations in #16806. I was [warned off using hasFeature like this](https://github.com/Automattic/wp-calypso/pull/16806#pullrequestreview-53891766) by @gwwar.

> [...] hasFeature checks should only be used for nudges at the moment, since these are client only checks mainly used for marketing/plans page.